src/com/hms_networks/americas/sc/realtimedata/RealTimeDataQueueManager.java

```diff
@@ -0,0 +1,141 @@
+package com.hms_networks.americas.sc.realtimedata;
+
+import java.util.ArrayList;
+
+import com.hms_networks.americas.sc.datapoint.DataPoint;
+import com.hms_networks.americas.sc.taginfo.TagGroup;
+import com.hms_networks.americas.sc.taginfo.TagInfo;
+import com.hms_networks.americas.sc.taginfo.TagInfoManager;
+
+/**
+ * This class is used to retrieve real time data from the Flexy.
+ *
+ * @author HMS Networks, MU Americas Solution Center
+ */
+public class RealTimeDataQueueManager {
+
+  /** Holds an array of lists of data points for every tag in group. */
+  private static ArrayList tagManagers;
+  /** Holds an array of the lists of what tags are in each group. */
+  private static ArrayList tagGroupList;
+  /**
+   * Lists of iterator for each tag group used to get a single point at a time from a tag group
+   * while retrieving data points.
+   */
+  private static int[] tagManagerIterators;
+  /** Index of information for tag group A in class ArrayLists. */
+  public static final int GROUP_A = 0;
+  /** Index of information for tag group B in class ArrayLists. */
+  public static final int GROUP_B = 1;
+  /** Index of information for tag group C in class ArrayLists. */
+  public static final int GROUP_C = 2;
+  /** Index of information for tag group D in class ArrayLists. */
+  public static final int GROUP_D = 3;
+  /** The number of tag groups. */
+  private static final int NUM_TAG_GROUPS = 4;

```

Comment:
what happened to the spacing in the constants?

Reply:
I personally prefer spaces all throughout the code whenever there is a small change in what the next code section is doing. this is more consistent now. I'll switch that back for the constants to give them room to breath.

Reply:
fixed

Reply:
It's actually a code convention: https://www.oracle.com/java/technologies/javase/codeconventions-whitespace.html